### ISSUE TITLE
Missing Layer normalization: no operator for variance (or mean_and_variance)

### ISSUE THREAD

**Issue by fhieber** [apache/incubator-mxnet#6995]
MXNet should support [Layer Normalization](https://arxiv.org/pdf/1607.06450.pdf) out-of-the-box, either by having an operator for it, or provide support operators that make it easier.
While it is possible to do with current MXNet ([see here](https://github.com/awslabs/sockeye/blob/master/sockeye/layers.py#L18)), it is a bit cumbersome, quite slow, and fairly tricky with shape inference due to the many broadcast_* ops.

In general it would be useful to have an operator to compute the variance, with a signature similar to ``mean``.
One could also think about  a ``mean_and_variance`` operator, similar to [tf.moments](https://www.tensorflow.org/api_docs/python/tf/nn/moments) that computes both statistics jointly which is likely more efficient.

Regarding a full Layer Normalization op it could look as follows:

LayerNormalization on dimension 3:
```
# output = scale * (inputs - mean) / sqrt(var + eps) + shift
output = mx.sym.LayerNormalization(data=input, axis=d3)
```
where the variables are created if not passed (similar to FullyConnected).

One should also be able to pass existing variables:
```
output = mx.sym.LayerNormalization(
  data=input,
  axis=d3,
  scale=mx.sym.Variable('scale'),
  shift=mx.sym.Variable('shift'),
  eps=1e-9)
```

In contrast to Batch normalization it is not necessary to compute running means or store anything from training for inference.

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to ping me to reopen if this is still an active issue. Thanks!
Also, do please check out our [forum](https://discuss.mxnet.io/) (and [Chinese version](https://discuss.gluon.ai/)) for general "how-to" questions.
